pkg/downstream/suggest.go

```diff
@@ -121,3 +122,58 @@ func iteratePullRequestCommits(ctx context.Context, client *github.Client, org,
 			return client.PullRequests.ListCommits(ctx, org, repo, prNum, o)
 		})
 }
+
+func hasCommit(git utils.GitHelper, req *SuggestRequest, found []string, c *github.Commit) (bool, error) {
+	has_commit := false
+	for _, commit := range found {
+		if len(commit) == 0 {
+			return false, nil
+		}
+
+		has, err := compareDiff(git, req, commit, c.GetURL())
+		if err != nil {
+			return false, err
+		}
+		has_commit = has
+	}
+
+	return has_commit, nil
+}
+
+func compareDiff(git utils.GitHelper, req *SuggestRequest, c string, u string) (bool, error) {
+	hash := strings.Split(u, "/commits/")[1]
+	url := "https://github.com/" + req.UpstreamOrg + "/" + req.UpstreamRepo + "/commit/" + hash + ".diff"
+	resp, err := http.Get(url)
+	if err != nil {
+		return false, err
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return false, err
+	}
+
+	remoteDiff := string(body)
+	localDiff, err := git.DoOutput("show", "--pretty=format:%n", c)
+	if err != nil {
+		return false, err
+	}
+	remoteDiffLines := strings.Split(strings.TrimSuffix(remoteDiff, "\n"), "\n")
+	localDiffLines := strings.Split(strings.TrimSuffix(localDiff, "\n"), "\n")
+
+	//remote diff sometimes contains an extra blank space as last line
+	if remoteDiffLines[len(remoteDiffLines)-1] == " " {
```

Comment:
Make this a for loop for both diffs, so that you're sure both are trimmed (in case github will start changing the format by adding a new line break, for example).

Also, please factorize the code into two distinct `pullDiff` and `compareDiffs` to divide the two logics